### PR TITLE
Remove template-haskell when running sql files so we don't need weird symlinks

### PR DESCRIPTION
**Pull request by ChrisPenner**
## Overview

Using template haskell to embed sql files has required a couple workarounds with symlinks, which for whatever reason stack doesn't like when building modules via a stack.yaml `git` dependency.

E.g.

```
Unsupported tarball from /tmp/nix-shell.4HGCSx/with-repo-archive24675/foo.tar: Invalid symbolic link from unison/codebase2/codebase-sqlite/unison to ., tried parsing unison/codebase2/codebase-sqlite/.: last component is a single dot
```

We can remove these symlinks if we switch to using data files, which allows us to depend on the `unison` project as a git dep.

## Implementation notes

Use `Paths_unison_codebase_sqlite.getDataFileName` to load and run sql migration files at run-time rather than embedding them in the executable using `hereFile`.

Had to remove the hack we used to prevent spurious Paths_*  module rebuilds in order to include this module.

## Test coverage

Tried both `stack install && unison -C new` and `stack exec unison -- -C new` and both seem to find and run `create.sql` just fine.

### CODE REVIEW

File: codebase2/codebase-sqlite/package.yaml [hunk 1]
```diff
@@ -2,12 +2,9 @@ name: unison-codebase-sqlite
 github: unisonweb/unison
 
 library:
-  when:
-    - condition: false
-      other-modules: Paths_unison_codebase_sqlite
```
Comment (ChrisPenner) [inline codebase2/codebase-sqlite/package.yaml:7]: I think this might now result in spurious re-builds, but obviously we need this module to exist.

**Comment by aryairani** [comment]
@ChrisPenner Awesome! But, how does this work for release builds? I'm assuming `stack install` is putting the data files somewhere in your home directory.

File: codebase2/codebase-sqlite/package.yaml (hunk 1 above)
Comment (aryairani) [reply codebase2/codebase-sqlite/package.yaml:7]: I think it still exists even with these lines there

**Comment by ChrisPenner** [comment]
Oh, good question, I think you're right.

I believe it would mean we couldn't distribute plain binaries any-more, which would be a bummer.

Presumably we could wire this up correctly in brew, but would probably need a proper installation method for every platform 😬 

I guess one other option is to just manually inline the sql as Text or w/e; which is annoying, but avoids both problems.

**Comment by aryairani** [comment]
Here's a related question but no answer yet: https://stackoverflow.com/questions/66215545/hardcode-datadir-path-upon-installing-executable

**Comment by aryairani** [comment]
Possibly relevant: https://cabal.readthedocs.io/en/stable/setup-commands.html#paths-in-the-simple-build-system

Possibly more relevant: [makeRelativeToProject](https://hackage.haskell.org/package/file-embed-0.0.15.0/docs/Data-FileEmbed.html#g:4) Does some path munging to try to guess where the root of the package is, before template-haskelling? Possibly avoiding the runtime dependency and avoiding the symlink nonsense

**Pull request closed without merging**